### ISSUE TITLE
We want to change both prefix list entries and a maximum number of entries in one request.

### ISSUE THREAD

**Issue by yamazoon0207** [hashicorp/terraform-provider-aws#26034]
We cannot modify "entry" and "max_entries" in a same request.

Sample Codes:

```
resource "aws_ec2_managed_prefix_list" "onpre" {
  name = var.pl_name

  address_family = "IPv4"
  max_entries    = length(var.onpremises_prefixes)

  dynamic "entry" {
    for_each = var.onpremises_prefixes
    content {
      cidr        = entry.value[0]
      description = entry.value[1]
    }
  }
}

variable "pl_name" { type = string }
variable "onpremises_prefixes" { type = list(list(string)) }
```

Errors:

```
Error: error updating EC2 Managed Prefix List (pl-xxxxx): InvalidParameterCombination: You cannot modify the entries and the maximum entries for the prefix list in the same request.
396    status code: 400, request id: xxxx
```

**Comment by justinretzolk** [comment]
Hey @yamazoon0207 👋 Thank you for taking the time to raise this! So that we have all of the necessary information in order to look into this, can you update the issue description to include all of the information requested in the [bug report template](https://github.com/hashicorp/terraform-provider-aws/blob/main/.github/ISSUE_TEMPLATE/Bug_Report.md?plain=1)?

**Comment by tjstansell** [comment]
Hm.. this seems pretty clear what the problem is.  The API doesn't allow the max size to change the same time as the entries list.  These need to be distinct steps.  So the provider should detect and increase the max size first, if that's being increased, then update the entries.  If the max size is decreasing, it should update the items first, then decrease the max size.  The AWS cli docs pretty clearly show this restriction:
```
       --max-entries (integer)
          The maximum number of entries for the prefix list. You cannot modify
          the entries of a prefix list and modify the size of a prefix list at
          the same time.

          If  any  of the resources that reference the prefix list cannot sup-
          port the new maximum size, the modify  operation  fails.  Check  the
          state  message  for  the  IDs of the first ten resources that do not
          support the new maximum size.
```

**Comment by scott-doyland-burrows** [comment]
I have an issue which I think is related.  I pull down a list of IPs and calculate the max entries from that list.  The resource is then created.  However, if I then manually remove an entry and then re-apply (which is pulling down the same list as in the first apply), I get the error below.

In my case I still have the same max entries set (to 15) in AWS and in terraform (terraform calculates the max entries).  Even if I manually set the max entries to a higher value than there are IPs, the issue still exists.

```
│ Error: Provider produced inconsistent final plan
│
│ When expanding the plan for aws_ec2_managed_prefix_list.cloudflare_ipv4 to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new value for
│ .entry: actual set element cty.ObjectVal(map[string]cty.Value{"cidr":cty.StringVal("172.64.0.0/13"), "description":cty.StringVal("cloudflare-ips-v4")}) does not correlate with any element in plan.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│
│ When expanding the plan for aws_ec2_managed_prefix_list.cloudflare_ipv4 to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new value for
│ .entry: block set length changed from 14 to 15.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

**Comment by yamazoon0207** [comment]
test

**Comment by justinretzolk** [comment]
Fixed by #26845
